[CORL-933] Add additional error handling around logQuery

nick-funk opened the pull request:
We are seeing that sometimes the context has an undefined logger when we log queries from GraphQL.

We want to catch this case and ensure we don't fail. If it happens again, we will catch everything we need to know and that will help us resolve the issue in the future.

nick-funk committed:
Add additional error handling around logQuery

We are seeing that sometimes the context has
an undefined logger. We probably want to catch
this case and ensure we don't fail. If it happens
again, we will catch everything we need to know
and that will help us resolve the issue in the
future.

CORL-933

wyattjoh commented on src/core/server/graph/extensions/LoggerExtension.ts:
```diff
@@ -33,9 +39,36 @@ export function logQuery(
+    if (!ctx || !ctx.logger) {
+      // Catch the context and log it out, we need to know what is
+      // going on here
+      logger.warn({ ctx }, "unable to find logger on GraphContext");
```
Avoid logging the context directly (it _will_ try to serialize everything) and instead just do a lighter reference:

```suggestion
      logger.warn({ ctx: Boolean(ctx) }, "unable to find logger on GraphContext");
```

wyattjoh commented on src/core/server/graph/extensions/LoggerExtension.ts:
```diff
@@ -33,9 +39,36 @@ export function logQuery(
+    // reasons, let's catch that to prevent other errors from
+    // occurring and log everything we can.
+    logger.error({ ctx, document }, "unable to log graphQL query");
+    logger.error(err);
```
Same feedback from above, as well as noting that the document node is also pretty heavy..

Noting as well that calling `logger.error(err)` does not work, and instead you should use:

```ts
logger.error({ err }, "unable to log GraphQL query");
```

Note the `{ err }` as an object. This is required because bunyan (our logger) uses the key names to assist with serialization, and passing `logger.error(err)` provides no keys.

nick-funk committed:
Simplify logging objects for efficiency

Some of the objects that we were going to log
are too large to be dumped to the logging context.
Instead, prefer simple boolean evaluation of their
existence.

CORL-933
